source/dofs/dof_tools_constraints.cc

```diff
@@ -1070,6 +1101,1316 @@ namespace DoFTools
+                       ++v)
+                    if (current_glob < this_face->child(child)->vertex_index(v))
+                      {
+                        current_max  = v;
+                        current_glob = this_face->child(child)->vertex_index(v);
+                      }
+
+                  if (current_max < 2)
+                    child_flip_y[child] = true;
+
+                  if (current_max % 2 == 0)
+                    child_flip_x[child] = true;
+
+                  if (this_face->child(child)->vertex_index(
+                        vertices_adjacent_on_face[current_max][0]) <
+                      this_face->child(child)->vertex_index(
+                        vertices_adjacent_on_face[current_max][1]))
+                    child_flip_xy[child] = true;
+
+                  child_flip_xy[child] = mother_flip_xy;
+                }
+
+              // copy the constraint matrix, since we need to modify that matrix
+              std::vector<std::vector<double>> constraints_matrix(
+                n_lines_on_children * fe.n_dofs_per_line() +
+                  n_of_children * fe.n_dofs_per_quad(),
+                std::vector<double>(dofs_on_mother.size(), 0));
+
+              { // copy the constraint matrix
+
+                for (unsigned int line = 0; line < 4; ++line)
+                  {
+                    unsigned int row_start   = line * fe.n_dofs_per_line();
+                    unsigned int line_mother = line / 2;
+                    unsigned int row_mother =
+                      (line_mother * 2) * fe.n_dofs_per_line();
+                    for (unsigned int row = 0; row < fe.n_dofs_per_line();
+                         ++row)
+                      for (unsigned int i = 0; i < 4 * fe.n_dofs_per_line();
+                           ++i)
+                        constraints_matrix[row + row_start][i] =
+                          fe.constraints()(row + row_mother, i);
+                  }
+
+                for (unsigned int line = 0; line < 4; ++line)
+                  {
+                    unsigned int row_start   = line * fe.n_dofs_per_line();
+                    unsigned int line_mother = line / 2;
+                    unsigned int row_mother =
+                      (line_mother * 2) * fe.n_dofs_per_line();
+                    for (unsigned int row = 0; row < fe.n_dofs_per_line();
+                         ++row)
+                      {
+                        for (unsigned int i = 4 * fe.n_dofs_per_line();
+                             i < dofs_on_mother.size();
+                             ++i)
+                          constraints_matrix[row + row_start][i] =
+                            fe.constraints()(row + row_mother, i);
+                      }
+                  }
+
+                // copy the weights for exterior edges
+                unsigned int row_offset = 4 * fe.n_dofs_per_line();
+                for (unsigned int line = 0; line < 8; line++)
```

Comment:
external lines?

Reply:
Yes, and similar to the above, I added a comment to clarify that. Furthermore, I replaced the "4" with n_external_lines_on_children to reduce the amount of magic numbers.